unit/java_bytecode/java_bytecode_parse_generics/parse_generic_superclasses.cpp

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing classes with generic superclasses with
+         unsupported signatures, falling back to using the raw type
+         descriptors
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_type.h>
+
+SCENARIO(
+  "parse generic superclass signature",
+  "[core][java_byte code[java_bytecode_parse_generics]]")
+{
+  const symbol_tablet &new_symbol_table = load_java_class(
+    "GenericBounds", "./java_bytecode/java_bytecode_parse_generics");
+
+  const std::string load_class("java::GenericBounds");
+  THEN("symbol is of non-generic class type")
+  {
+    const symbolt &class_symbol = new_symbol_table.lookup_ref(load_class);
+
+    const typet &class_type = class_symbol.type;
+    REQUIRE(class_type.id() == ID_struct);
+
+    const struct_typet &class_struct = to_struct_type(class_type);
+
+    THEN("it has two fields called upper and lower")
+    {
+      const struct_union_typet::componentt &field_upper =
+        require_type::require_component(class_struct, "gen_upper");
+      const struct_union_typet::componentt &field_lower =
+        require_type::require_component(class_struct, "gen_lower");
+      const struct_union_typet::componentt &field_interface =
+        require_type::require_component(class_struct, "gen_interface");
+
+      THEN(
+        "these fields have a non-generic base class / interface  as their real "
+        "generic signature is unsupported at the moment")
+      {
+        // once bounds in generic signatures are supported, this test must be
+        // changed to check for the correct generic types
+        const symbolt &upper_symbol =
+          new_symbol_table.lookup_ref("java::GenericBoundsUpper");
```

Comment:
I think everything from line 22 until here can be deleted, we don't really need to check for the type or fields of `GenericBounds`. The way I understand it is that that class is only a wrapper to bring in all those interesting classes with generic superclasses/interaces at the same time.